server/src/main/java/org/elasticsearch/action/admin/indices/create/CreateIndexRequest.java

```diff
@@ -255,11 +248,10 @@ public CreateIndexRequest cause(String cause) {
     /**
      * Adds mapping that will be added when the index gets created.
      *
-     * @param type   The mapping type
      * @param source The mapping source
      */
-    public CreateIndexRequest mapping(String type, XContentBuilder source) {
-        return mapping(type, BytesReference.bytes(source), source.contentType());
+    public CreateIndexRequest mapping(XContentBuilder source) {
```

Comment:
For consistency with the changes in `CreateIndexRequestBuilder`, it would be great if we also removed the type parameter from `public CreateIndexRequest mapping(String type, Map<String, ?> source)`, and converted as many callers as possible to use the new signature. Here's a suggestion for how to do it:

```

    public CreateIndexRequest mapping(Map<String, ?> source) {
        return mapping(MapperService.SINGLE_MAPPING_NAME, source);
    }

    private CreateIndexRequest mapping(String type, Map<String, ?> source) {
        // wrap it in a type map if its not
        if (source.size() != 1 || !source.containsKey(type)) {
            source = Map.of(MapperService.SINGLE_MAPPING_NAME, source);
        }
        ...
    }
```

The private method is still needed to support `CreateIndexRequest#source`.

Reply:
++